Remotion/ObjectBinding/Web/UI/Controls/BocListImplementation/Validation/IBocListValidationFailureHandler.cs

```diff
@@ -0,0 +1,54 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// using NUnit.Framework;
+//
+
+using System;
+using Remotion.ObjectBinding.Web.UI.Controls.Validation;
+
+namespace Remotion.ObjectBinding.Web.UI.Controls.BocListImplementation.Validation
+{
+  /// <summary>
+  /// The <see cref="IBocListValidationFailureHandler"/> interface provides functionality to handle validation failures.
+  /// </summary>
+  /// <remarks>
+  ///   New inheritors of the <see cref="IBocListValidationFailureHandler"/> interface should have a position between -10e9 and 10e9.
+  /// </remarks>
+  /// <seealso cref="BocListListValidationFailureHandler"/>
+  /// <seealso cref="BocListRowAndCellValidationFailureHandler"/>
+  public interface IBocListValidationFailureHandler
+  {
+    /// <summary>
+    /// Handles validation failures.
+    /// </summary>
+    /// <remarks>
+    ///   Implementations of this method should handle some part of the validation failures belonging to the
+    ///   <see cref="T:Remotion.ObjectBinding.Web.UI.Controls.BocListImplementation.IBocList"/> found in its
+    ///   <see cref="IBocListValidationFailureRepository"/>.
+    ///
+    ///   Use <paramref name="context"/> to access the <see cref="T:Remotion.ObjectBinding.Web.UI.Controls.BocListImplementation.IBocList"/>
+    ///   and <see cref="IBocListValidationFailureRepository" /> and for reporting the relevant validation failures.
+    ///
+    ///   All unhandled failures will be handled by the <see cref="BocListRowAndCellValidationFailureHandler"/>,
+    ///   unless it has been removed.
+    ///   There will be no <see cref="T:Remotion.ObjectBinding.Web.UI.Controls.BocListImplementation.IBocList"/>
+    ///   business object property validation failures in the <see cref="IBocListValidationFailureRepository"/>
+    ///   when calling this method on your implementation,
+    ///   unless the <see cref="BocListListValidationFailureHandler"/> has been removed or your implementation is out of order in the dependency container.
```

Comment:
Not sure about the meaning of this last sentence.

still incorrect, but not the reason I did not understand it:
"out of order" does not mean "incorrect sequence" but means "the machine does not work"

Maybe you can say it in a posible way: For there to be validation failures, you need to...?

Reply:
For the <see cref="IBocListValidationFailureRepository"/> to contain any <see cref="IBocList"/> business object property validation failures your implementation has to be called before the <see cref="BocListListValidationFailureHandler"/> (provided the service has not been removed).